src/include/network/routines.rb

```diff
@@ -742,24 +756,62 @@ def SetIfaceDown(dev_name)
 
     # Tries to set all interfaces up
     #
-    # @return [boolean] false if some of interfaces cannot be set up
+    # @return [boolean] false if interfaces are empty or some
+    # interface cannot be bring up
     def SetAllIfacesUp
+      all_up = true
       interfaces = GetAllInterfaces()
 
       return false if interfaces.empty?
 
-      interfaces.all? { |i| SetIfaceUp(i) }
+      interfaces.each { |i| all_up = false if !SetIfaceUp(i) }
+
+      all_up
```

Comment:
My proposal is: if a call fails, we cannot react to it anyway. Discard the return value and let the `SetAll*` methods become oneliners:

```rb
# Tries to set all interfaces up. Errors are logged.
# @return [void]
def SetAllIfacesUp
  GetAllInterfaces().each { |i| SetIfaceUp(i) }
end
```